maven-extension/README.md

```diff
@@ -175,6 +175,71 @@ The [Jenkins OpenTelemetry Plugin](https://plugins.jenkins.io/opentelemetry/) ex
 
 The [`otel-cli`](https://github.com/equinix-labs/otel-cli) is a command line wrapper to observe the execution of a shell command as an OpenTelemetry trace.
 
+## Instrumenting Maven Mojos for better visibility in Maven builds
+
+Maven plugin authors can instrument Mojos for better visibility in Maven builds.
+
+Common instrumentation patterns include:
+
+* Adding contextual data as attributes on the spans created by the OpenTelemetry Maven Extension,
+* Creating additional sub spans to breakdown long mojo goal executions in finer grained steps
+
+Note that the instrumentation of a plugin is enabled when the OpenTelemetry Maven extension is added to the build and activated.
+Otherwise, the instrumentation of the Maven plugin is noop.
+
+It is recommended to enrich spans using the [OpenTelemetry Semantic Conventions](https://opentelemetry.io/docs/concepts/semantic-conventions/)
+to improve the visualization and analysis inObservability products.
+The [HTTP](https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/http/)
+and [database client calls](https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/database/)
+conventions are particularly useful when  invoking external systems.
+
+Steps to instrument a Maven Mojo:
+
+* Add the OpenTelemetry API dependency in the `pom.xml` of the Maven plugin.
+
+```xml
+<project>
+    ...
+    <dependencies>
+        <dependency>
+            <groupId>io.opentelemetry</groupId>
+            <artifactId>opentelemetry-api</artifactId>
+            <version>1.23.0</version>
```

Comment:
Should we set the version to a parameter to allow for future releases and not causing confusion about the version to use?

Reply:
```suggestion
            <version>LATEST</version>
```